packages/flutter_driver/lib/src/common/diagnostics_tree.dart

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'enum_util.dart';
+import 'find.dart';
+import 'message.dart';
+
+/// [DiagnosticsNode] tree types that can be requested by [GetDiagnosticsTree].
+enum DiagnosticsType {
+  /// The [DiagnosticsNode] tree formed by [RenderObject]s.
+  renderObject,
+
+  /// The [DiagnosticsNode] tree formed by [Widget]s.
+  widget,
+}
+
+EnumIndex<DiagnosticsType> _diagnosticsTypeIndex = EnumIndex<DiagnosticsType>(DiagnosticsType.values);
+
+/// A Flutter Driver command to retrieve the json-serialized [DiagnosticsNode]
+/// tree of the object identified by [finder].
+///
+/// The [DiagnosticsType] of the [DiagnosticsNode] tree returned is specified by
+/// [diagnosticsType].
+class GetDiagnosticsTree extends CommandWithTarget {
+  /// Creates a [GetDiagnosticsTree] Flutter Driver command.
+  GetDiagnosticsTree(SerializableFinder finder, this.diagnosticsType, {
+    this.subtreeDepth = 0,
+    this.includeProperties = true,
+    Duration timeout,
+  }) : super(finder, timeout: timeout);
+
+  /// Deserializes this command from the value generated by [serialize].
+  GetDiagnosticsTree.deserialize(Map<String, dynamic> json)
+      : subtreeDepth = json['subtreeDepth'] == 'null' ? null : int.parse(json['subtreeDepth']),
+        includeProperties = json['includeProperties'] == 'true',
+        diagnosticsType = _diagnosticsTypeIndex.lookupBySimpleName(json['diagnosticsType']),
+        super.deserialize(json);
+
+  /// How many levels of children to include in the JSON result.
+  ///
+  /// Defaults to zero, which will only return the [DiagnosticsNode] information
```

Comment:
Says that it defaults to zero, but it looks like it can be null above?

Reply:
That null handling as a left-over from something else I tried. I am going to remove it and add a non-null assert.